app/models/concerns/package_filtering.rb

```diff
@@ -61,6 +63,11 @@ def where_states(states)
       where(sql_query)
     end
 
+    def filter_by_loss_states(states)
+      states = "'#{states.join("','")}'"
+      joins(:package_actions).where("packages_inventories.action in (#{states})")
```

Comment:
This leads to SQL injection issues. Always write queries with parameters so that ActiveRelation will automatically sanitize them for you: 

```
where("packages_inventories.action IN (:terms)", terms: states)
```